src/app.rs

```diff
@@ -1,110 +1,148 @@
+            arg("color")
                 .long("color")
                 .short("c")
                 .takes_value(true)
                 .value_name("when")
                 .possible_values(&["never", "auto", "always"])
-                .hide_possible_values(true)
-                .help(
-                    "When to use color in the output:\n\
-                     never, auto, always (default: auto)",
-                ),
+                .hide_possible_values(true),
         )
         .arg(
-            Arg::with_name("threads")
+            arg("threads")
                 .long("threads")
                 .short("j")
                 .takes_value(true)
-                .value_name("num")
-                .help(
-                    "Set number of threads to use for searching\n\
-                     (default: number of available CPU cores)",
-                ),
+                .value_name("num"),
         )
         .arg(
-            Arg::with_name("max-buffer-time")
+            arg("max-buffer-time")
                 .long("max-buffer-time")
                 .takes_value(true)
-                .hidden(true)
-                .help("the time (in ms) to buffer, before streaming to the console"),
+                .hidden(true),
         )
-        .arg(Arg::with_name("pattern").help("the search pattern, a regular expression (optional)"))
-        .arg(Arg::with_name("path").help("the root directory for the filesystem search (optional)"))
+        .arg(arg("pattern"))
+        .arg(arg("path"))
+}
+
+#[cfg_attr(rustfmt, rustfmt_skip)]
+fn usage() -> HashMap<&'static str, Help> {
+    let mut h = HashMap::new();
+    doc!(h, "hidden"
+        , "Search hidden files and directories"
+        , "Includes, in a search query, any hidden files and directories for the directory of \
+            the search.");
+    doc!(h, "no-ignore"
+        , "Do not respect .(git)ignore files"
+        , "Searches for a pattern match with files and directories, including all defined \
+            `.*ignore` file exceptions. Useful when you want to match inclusively for a pattern.");
+    doc!(h, "case-sensitive"
+        , "Case-sensitive search (default: smart case)"
+        , "Define your pattern as case sensitive. By default, fd uses 'smart case' for queries.\
+            This option disables smart case.");
+    doc!(h, "absolute-path"
+        , "Show absolute instead of relative paths"
+        , "Shows the result of a pattern match as an absolute path instead of relative path.");
+    doc!(h, "follow"
+        , "Follow symbolic links"
+        , "Includes a symbolic links destination in the pattern match.");
+    doc!(h, "full-path"
+        , "Search full path (default: file-/dirname only)"
+        , "Searches the full path of directory. By default, fd only navigates file-/dirname.");
+    doc!(h, "null_separator"
+        , "Separate results by the null character"
+        , "Separates each pattern match result by the null character.");
```

Comment:
We could refer to `xargs` here: "Separate search results by the null character (instead of newlines). This is useful for piping results to 'xargs'".